Ignore Recovery Partitions

Chirishman opened the pull request:
**Pull Request (PR) description**
Filtering out 'Recovery' type partitions from xvhdfile method pipelines.

Adding specification of a numbered partition can be enhanced under a separate cover if desired, but I can't see a scenario where anyone is intentionally injecting contents into a VM's recovery partition so it seems safe to filter out that partition type.

**This Pull Request (PR) fixes the following issues:**
Fixes #134
Partially Fixes #27 
Partially Fixes #26 
Partially Fixes #59

**Task list:**
- [x] Filter Recovery partitions out of the pipeline, resolving the terminating error
- [ ] Update documentation/samples to make users aware of the current partition-based limitations of xVhdFile
- [ ] Add manual partition selection parameter

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/xhyper-v/135)
<!-- Reviewable:end -->

Chirishman commented:
Man, none of those failures are mine except using `?` instead of `Where-Object`.

All of the rest of that is in the original commit. I can fix it if you really need me to do so to accept the pull request, but you guys are using `%` everywhere instead of `Foreach-Object` so I don't really feel like it's my responsibility to own every instance where you guys put `$null` on the right of the compare instead of the left and neglected to capitalize your [Parameter()] tags.

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -9,14 +9,15 @@ function Get-TargetResource
     param
     (
-        [parameter(Mandatory = $true)]
+        [Parameter(Mandatory = $true)]
```
Fixed capitalization to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -9,14 +9,15 @@ function Get-TargetResource
         $VhdPath,
 
-        [parameter(Mandatory = $true)]
+        [Parameter(Mandatory = $true)]
```
Fixed capitalization to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -9,14 +9,15 @@ function Get-TargetResource
         [Microsoft.Management.Infrastructure.CimInstance[]]
         $FileDirectory,
 
+        [Parameter()]
```
Added to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -75,14 +76,15 @@ function Set-TargetResource
         $VhdPath,
 
-        [parameter(Mandatory = $true)]
+        [Parameter(Mandatory = $true)]
```
Fixed capitalization to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -75,14 +76,15 @@ function Set-TargetResource
         $FileDirectory,
-
+        
+        [Parameter()]
```
Added to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -158,14 +160,15 @@ function Test-TargetResource
     param
     (
-        [parameter(Mandatory = $true)]
+        [Parameter(Mandatory = $true)]
```
Fixed capitalization to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -158,14 +160,15 @@ function Test-TargetResource
         $VhdPath,
 
-        [parameter(Mandatory = $true)]
+        [Parameter(Mandatory = $true)]
```
Fixed capitalization to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -158,14 +160,15 @@ function Test-TargetResource
         $FileDirectory,
-
+        
+        [Parameter()]
```
Added to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -250,7 +253,7 @@ function Test-TargetResource
             if ($itemToCopy.Attributes)
             {
-                $currentAttribute = @(Get-ItemProperty -Path $finalDestinationPath |% Attributes)
+                $currentAttribute = @(Get-ItemProperty -Path $finalDestinationPath | ForEach-Object -MemberName Attributes)
```
Replaced alias to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -271,11 +274,11 @@ function EnsureVHDState
     param(        
         
-        [parameter(Mandatory=$false,ParameterSetName = "Mounted")]
+        [Parameter(ParameterSetName = "Mounted")]
```
Fixed capitalization to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -271,11 +274,11 @@ function EnsureVHDState
+        [Parameter(ParameterSetName = "Mounted")]
         [switch]$Mounted,
-        [parameter(Mandatory=$false,ParameterSetName = "Dismounted")]  
+        [Parameter(ParameterSetName = "Dismounted")]  
```
Fixed capitalization to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -271,11 +274,11 @@ function EnsureVHDState
+        [Parameter(ParameterSetName = "Dismounted")]  
         [switch]$Dismounted,
-        [parameter(Mandatory=$true)]
+        [Parameter(Mandatory=$true)]
```
Fixed capitalization to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -312,52 +315,69 @@ function EnsureVHDState
-            $returnValue.Type = 'Directory'
-         }
+    param(
+        [Parameter()]
```
Added to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -367,19 +387,21 @@ function SetVHDFile
      [CmdletBinding(DefaultParametersetName="Copy")] 
     param(       
-        [parameter(Mandatory=$true,ParameterSetName = "Copy")]
+        [Parameter(Mandatory=$true,ParameterSetName = "Copy")]
```
Fixed capitalization to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -367,19 +387,21 @@ function SetVHDFile
-        [parameter(Mandatory=$true,ParameterSetName = "Copy")]
+        [Parameter(Mandatory=$true,ParameterSetName = "Copy")]
         $sourcePath,        
+        [Parameter()]
```
Added to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -367,19 +387,21 @@ function SetVHDFile
         $sourcePath,        
+        [Parameter()]
         [switch]$recurse,
+        [Parameter()]
```
Added to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -367,19 +387,21 @@ function SetVHDFile
+        [Parameter()]
         [switch]$force,
-        [parameter(Mandatory=$false,ParameterSetName = "New")]  
+        [Parameter(ParameterSetName = "New")]  
```
Fixed capitalization and removed property to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -367,19 +387,21 @@ function SetVHDFile
+        [Parameter(ParameterSetName = "New")]  
         $type,
-        [parameter(Mandatory=$false,ParameterSetName = "New")]  
+        [Parameter(ParameterSetName = "New")]  
```
Fixed capitalization and removed property to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -367,19 +387,21 @@ function SetVHDFile
+        [Parameter(ParameterSetName = "New")]  
         $content,       
-        [parameter(Mandatory=$true)]
+        [Parameter(Mandatory=$true)]
```
Fixed capitalization to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -367,19 +387,21 @@ function SetVHDFile
+        [Parameter(Mandatory=$true)]
         $destinationPath, 
-        [parameter(Mandatory=$true,ParameterSetName = "Set")]  
+        [Parameter(Mandatory=$true,ParameterSetName = "Set")]  
```
Fixed capitalization to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -367,19 +387,21 @@ function SetVHDFile
+        [Parameter(Mandatory=$true,ParameterSetName = "Set")]  
         $attribute,
-        [parameter(Mandatory=$true,ParameterSetName = "Delete")]
+        [Parameter(Mandatory=$true,ParameterSetName = "Delete")]
```
Fixed capitalization to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -417,13 +439,13 @@ function SetVHDFile
 {
     param(
-    [parameter(Mandatory=$true)]
+    [Parameter(Mandatory=$true)]
```
Fixed capitalization to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -417,13 +439,13 @@ function SetVHDFile
     [ValidateScript({Test-Path $_})] 
     $sourcePath,
-    [parameter(Mandatory=$true)]
+    [Parameter(Mandatory=$true)]
```
Fixed capitalization to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -417,13 +439,13 @@ function SetVHDFile
     [ValidateScript({Test-Path $_})]
     $destinationPath,
-    [parameter(Mandatory=$false)]
+    [Parameter()]
```
Fixed capitalization and removed property to please AppVeyor validation

Chirishman commented on DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1:
```diff
@@ -312,52 +315,69 @@ function EnsureVHDState
 # Change the Cim Instance objects in to a hash table containing property value pair.
 function GetItemToCopy
 {
-    param([Microsoft.Management.Infrastructure.CimInstance] $item)
```
Got sick of looking at this repetitive code as I was staring at it, fixing the validation errors. I decided it was better to rewrite it with loops instead of just swapping the $null references to the left side of the comparison.

Chirishman commented:
@joeyaiello Who  is maintaining this repo?

BrianFarnhill commented:
@Chirishman Same as the other PR, the only failing tests here are from that mock command, if you do a pull from PowerShell/dev you'll get the changes and then the tests should light up and I'll review the rest of this one for you.

Chirishman commented:
@BrianFarnhill Done

BrianFarnhill commented:
One minor style change but the rest looks good. Clever use of that ordered list to address the order of pulling those properties out too by the way, I like it :) You'll also need to do a pull from PowerShell/dev to resolve a couple of conflicts as well

---

Reviewed 1 of 1 files at r2, 5 of 5 files at r3.
Review status: all files reviewed at latest revision, 26 unresolved discussions, some commit checks failed.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 350 at r3](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L0kozGg_jLOtPNKWzUL:-L0kozGhD1EZ3WAvb4QL:b-s2os8) ([raw file](https://github.com/powershell/xhyper-v/blob/3afe55e4813d834feeabece0142ab7a686f8eef7/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L350)):*
> ```PowerShell
>                 #Special behavior for the Type property based on SourcePath
>                 #This relies on SourcePath preceeding Type in the list of keys (the reason for using OrderedDictionary)
>                 if (Test-Path $returnValue.SourcePath -PathType Leaf ) {
> ```

Style guideline thing, can you bump the { to a new line beneath the if statement?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-:-L0kpgCvAo4rQgcT3TMj:bg22agc)*
<!-- Sent from Reviewable.io -->

Chirishman commented:
*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 350 at r3](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L0kozGg_jLOtPNKWzUL:-L0tron9gNJIRp0e_p41:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/3afe55e4813d834feeabece0142ab7a686f8eef7/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L350)):*
<details><summary><i>Previously, BrianFarnhill (Brian Farnhill) wrote…</i></summary><blockquote>

Style guideline thing, can you bump the { to a new line beneath the if statement?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/135)*
<!-- Sent from Reviewable.io -->

Chirishman commented:
Fixed. Thanks! I'm a big fan of solving that kind of problem with hashtables or ordered dictionaries instead of endless repetitive if/elseif statements.

---

Review status: 0 of 1 files reviewed at latest revision, 26 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-:-L0tsakq--8kzCHWXdmX:beyoeaz)*
<!-- Sent from Reviewable.io -->

Chirishman commented:
Review status: 0 of 1 files reviewed at latest revision, 26 unresolved discussions.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 12 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-12:-L0tuaV5Yy_cEv2ujXT3:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L12)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Fixed capitalization to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 16 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-16:-L0tubetcrU9d3g7-mzi:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L16)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Fixed capitalization to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 20 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-20:-L0tuc1ZdvArN5JfAcSv:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L20)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Added to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 79 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-79:-L0tucaimcjMMLZ805y2:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L79)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Fixed capitalization to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 83 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-83:-L0tud6ptZ_zes20BHzC:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L83)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Fixed capitalization to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 87 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-87:-L0tudURI480J9ymWiww:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L87)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Added to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 163 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-163:-L0tudybvTTQTceZxVH6:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L163)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Fixed capitalization to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 167 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-167:-L0tuedAAc3uijK2GoNP:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L167)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Fixed capitalization to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 171 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-171:-L0tufEr0fx1rBRqVbgd:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L171)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Added to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 256 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-256:-L0tufgwJULjU5KDeSbz:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L256)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Replaced alias to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 277 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-277:-L0tugNO_hViP11xtC9O:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L277)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Fixed capitalization and removed property to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 279 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-279:-L0tugjZ76NEdVXweNZG:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L279)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Fixed capitalization and removed property to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 281 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-281:-L0tuh7NJP0yM6ibTIlV:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L281)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Fixed capitalization to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 319 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-319:-L0tuhYWz5ZkmNrD1jOI:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L319)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Added to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 390 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-390:-L0tuiHt9oCNjJ0uMm7D:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L390)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Fixed capitalization to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 392 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-392:-L0tuikO8AezsthncJHX:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L392)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Added to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 394 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-394:-L0tuj1HHQj9ARowsuxp:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L394)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Added to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 396 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-396:-L0tujMmubdIdNCSz966:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L396)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Fixed capitalization and removed property to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 398 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-398:-L0tujh-gxqpokgiMlMh:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L398)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Fixed capitalization and removed property to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 400 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-400:-L0tuk8-eph6EdW_KXNj:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L400)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Fixed capitalization to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 402 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-402:-L0tukUBZh2bEvGrrstO:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L402)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Fixed capitalization to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 404 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-404:-L0tukvTpsp-S1ABSkGm:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L404)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Fixed capitalization to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 442 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-442:-L0tulKa8WzMrzGX-_Yf:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L442)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Fixed capitalization to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 445 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-445:-L0tulftWZ0kN6FXSLNo:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L445)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Fixed capitalization and removed property to please AppVeyor validation
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1, line 448 at r2](https://reviewable.io:443/reviews/powershell/xhyper-v/135#-L-bDCCp3Kw13rP2qOkC-r2-448:-L0tumA3jaqJWstpDOzl:b-896fix) ([raw file](https://github.com/powershell/xhyper-v/blob/4614ae0400745d9b61c9c9ce492a17b99d900e0a/DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1#L448)):*
<details><summary><i>Previously, Chirishman wrote…</i></summary><blockquote>

Fixed capitalization and removed property to please AppVeyor validation
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/135)*
<!-- Sent from Reviewable.io -->

Chirishman commented:
@BrianFarnhill Happy New Year

BrianFarnhill commented:
@Chirishman Back at you - I haven't forgotten about this, just been away with family over the holidays. I'll be back in full swing next week, I'll make sure this gets done as soon as I can at that point.

BrianFarnhill commented:
Reviewed 1 of 4 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xhyper-v/135)*
<!-- Sent from Reviewable.io -->
